src/API/WorkController.ts

```diff
@@ -26,9 +30,11 @@ export class WorkController {
     return this.collection.findOne({ id }, { fields: { _id: false } })
   }
 
-  async getByPublicKey(publicKey: string): Promise<any> {
-    this.logger.trace({ method: 'getByPublicKey', publicKey }, 'Getting Works by Public Key from DB')
-    return this.collection.find({ publicKey }, { fields: { _id: false } }).toArray()
+  async getByFilters(worksFilters: WorksFilters = {}): Promise<any> {
+    const definedFilters = Object.entries(worksFilters)
+      .filter(([key, value]) => value !== undefined)
+      .toObject()
+    return this.collection.find(definedFilters, { fields: { _id: false } }).toArray()
```

Comment:
Are there any security issues or misuses that could arise from passing entire query object into collection.find()? I am thinking it might be better to explicity pass in filters we accept, and maybe a test case should be created to avoid this.

Reply:
@lautarodragan @warrenv ping, I think @jwicks31 might be awaiting your feedback here.

Reply:
Thanks Kenny. 

Joi is currently validating that no unexpected fields arrive in the API layer, so nothing unexpected should reach this function.

Ultimately the controller layer shouldn't depend other layers "behaving nice", so to speak. It should be atomically functional, including any validation it wants to do. 

But for now I'm OK doing it this way — it's not the only place this layer depends on other layers for validation and not an immediate issue. Maybe we could create an issue?

Reply:
I agree with @lautarodragan - it's ok for now but it is a potential crack for sql injection.
I'd like to see a unit test put in place for it in this PR at least.

Reply:
This PR does have a test to ensure that if we pass a filter that isn't accepted we get the correct response. I think I may be misunderstanding though. Below is the test and the helper function.

```ts
  @AsyncTest()
  @TestCase('test')
  async getWorksShouldFailWith422WhenPassingAnInvalidArgument(test: string) {
    const response = await this.client.getWorksTest(test)

    Expect(response.status).toBe(422)
  }
```
```ts
  getWorksTest(test?: string) {
    return fetch(this.url + '/works/?test=' + test)
  }
```